### PR TITLE
Introduce 'git backfill' to get missing blobs in a partial clone

### PR DESCRIPTION
**Pull request by derrickstolee**
This change introduces the `git backfill` command which uses the path walk API to download missing blobs in a blobless partial clone.

By downloading blobs that correspond to the same file path at the same time, we hope to maximize the potential benefits of delta compression against multiple versions.

These downloads occur in a configurable batch size, presenting a mechanism to perform "resumable" clones: `git clone --filter=blob:none` gets the commits and trees, then `git backfill` will download all missing blobs. If `git backfill` is interrupted partway through, it can be restarted and will redownload only the missing objects.

When combining blobless partial clones with sparse-checkout, `git backfill` will assume its `--sparse` option and download only the blobs within the sparse-checkout. Users may want to do this as the repo size will still be smaller than the full repo size, but commands like `git blame` or `git log -L` will not suffer from many one-by-one blob downloads.

Future directions should consider adding a pathspec or file prefix to further focus which paths are being downloaded in a batch.

### CODE REVIEW

File: builtin/backfill.c [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#include "strmap.h"
+#include "string-list.h"
+#include "revision.h"
+#include "trace2.h"
+#include "progress.h"
+#include "packfile.h"
+#include "path-walk.h"
+
+static const char * const builtin_backfill_usage[] = {
+	N_("git backfill [--batch-size=<n>] [--[no-]sparse]"),
```
Comment (derrickstolee) [inline builtin/backfill.c:26]: In before @dscho: this needs an `(EXPERIMENTAL!) ` flag.

**Review by dscho — APPROVED** [review]
Great stuff! Most of the patches I saw before, in one form or other, so there were no surprises for me.

I've added one more commit (to mark the command as experimental), and plan on merging this tomorrow (so that we get a new [snapshot](https://wingit.blob.core.windows.net/files/index.html), with a brand-new MSYS2 runtime, no less!).

**Comment by derrickstolee** [comment]
Thanks, @dscho. One more big one to go!

**Comment by dscho** [comment]
/add relnote feature The new, experimental `git backfill` command [was added](https://github.com/git-for-windows/git/pull/5172): It helps fetching relevant Git objects smartly in a partial, sparse clone.

**Pull request merged**